open3d_conversions/src/open3d_conversions.cpp

```diff
@@ -15,9 +15,45 @@
 // C++
 #include <memory>
 #include <string>
+#include <sstream>
 
+#include <sensor_msgs/image_encodings.hpp>
 #include "open3d_conversions/open3d_conversions.hpp"
 
+// This is the best we have prior to C++20 std::endian::native
+static bool isLittleEndian(){
+  const int32_t value = 0x01;
+  const std::byte * least_significant_address = 
+    reinterpret_cast<const std::byte *>(&value);
+  return (*least_significant_address == std::byte{0x01});
```

Comment:
Curious: where did you get this code from or did you write this? Just something I can glance over and assume is right if you grabbed it from somewhere else in ROS

Reply:
I had found this code on StackOverflow as a non-UB version of checking endianess, but your comment here encouraged me to check how they do it in the cv_bridge package and apparently there's a `<rcpputils/endian.hpp>` header which behaves very similarly to C++ 20 std::endian, so I'll replace this function with that.